lib/modules/dosomething/dosomething_helpers/dosomething_helpers.share.inc

```diff
@@ -109,6 +111,11 @@ function dosomething_helpers_utm_share_paths($share_types, $path, $utm_campaign)
       )
     );
     $share_paths[$type] = $path . $utm_params;
+
+    // Append the custom share source to the URL if we have one
+    if ($custom_share_source) {
+      $share_paths[$type] .= '&source=' . $custom_share_source;
```

Comment:
I think the `&` will get encoded strange here, might be best to pass this through `http_build_query` like `dosomething_helpers_utm_parameters` does